### ISSUE TITLE
element.detach in SAXReader's ElementHandler is very slow

### ISSUE THREAD

**Issue by wlinna** [dom4j/dom4j#121]
Hello,
I'm trying to follow the example (from [FAQ](https://github.com/dom4j/dom4j/wiki/FAQ)) on handling very large XML documents. However, even with relative small files (80 MiB) the method is  very slow. Here's my code:

```
        SAXReader saxReader = new SAXReader(false);
  
        saxReader.addHandler("/LandXML/Surfaces/Surface/Definition/Pnts/P", new ElementHandler() {
            @Override
            public void onStart(ElementPath elementPath) {}

            @Override
            public void onEnd(ElementPath elementPath) {
                org.dom4j.Element node = elementPath.getCurrent();
                //  
                node.detach();
            }
        });

        org.dom4j.Document document = saxReader.read(is);
```
My test file is here:
https://drive.google.com/file/d/1duZydaOlhFAeEB1O41rAwWip8ad9tvAo/view?usp=sharing

If I comment out `node.detach()`, the code seems to complete in 10-20 seconds, otherwise it takes at least 3 minutes to read. 

My schedule is quite tight. Is there a workaround I could try? I considered detaching Pnts element after all P elements have been removed, but then RAM might become an issue.

Thanks

**Comment by wlinna** [comment]
Thanks, but I actually want to process P and F elements. My workaround is to remove Pnts and Faces elements when their end event is raised. It's not an optimal solution, but it's enough to let me deal with other bottlenecks.

**Comment by reinhapa** [comment]
@wlinna if you do not need a document you could also only process the document using a `SAXParser` reacting on the `P` and  `F` elements.

**Comment by wlinna** [comment]
Unfortunately, I need to hand out the rest of the document to a DOM based reader (not written by us). Then I just insert our separately handled P and F elements to the data model.
